src/index.ts

```
@@ -18,6 +24,40 @@ const buildRestClient = (apiClient: InfisicalApi, requestOptions?: RawAxiosReque
+
+		if (!config._retryCount) config._retryCount = 0;
+
+		// handle rate limits and network errors
+		if ((error.response?.status === 429 || error.response?.status === undefined) && config && config._retryCount! < maxRetries) {
+			config._retryCount!++;
+			const exponentialDelay = Math.min(initialRetryDelay * Math.pow(backoffFactor, config._retryCount! - 1));
+
+			return new Promise(async resolve => {
+				await new Promise(resolve => setTimeout(resolve, exponentialDelay));
+				resolve(axiosInstance(config));
+			});
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)